api/islandora_xacml_api.drush.inc

```diff
@@ -135,3 +157,144 @@ EOQ;
+      $context['finished'] = 1;
+      $context['message'] = t('Nothing to verify.');
+      return;
+    }
+    $sandbox['result_stash'] = array();
+  }
+  if (empty($sandbox['result_stash'])) {
+    if (isset($sandbox['last_date'])) {
+      $filter = format_string('FILTER(?created > "!created"^^xsd:dateTime || (?created = "!created"^^xsd:dateTime && xsd:string(?object) > "info:fedora/!pid"^^xsd:string))', array(
+        '!created' => $sandbox['last_date'],
+        '!pid' => $sandbox['last_pid'],
+      ));
+    }
+    $limited_query = format_string($query, array(
+      '!filter' => isset($filter) ? $filter : '',
+    )) . <<<EOQ
+LIMIT $limit
+EOQ;
+
+    drush_log(dt('Querying for 100 objects.'));
+    $sandbox['result_stash'] = $tuque->repository->ri->sparqlQuery($limited_query);
+    if (empty($sandbox['result_stash'])) {
+      $context['finished'] = 1;
+      $context['message'] = t('Updated @count PID(s) and ran out of items early. POLICY was manually removed?', array(
+        '@count' => $sandbox['progress'],
+      ));
+      return;
+    }
+  }
+
+  $result = array_shift($sandbox['result_stash']);
+  $sandbox['last_date'] = $result['created']['value'];
+  $sandbox['last_pid'] = $result['object']['value'];
+  $object = islandora_object_load($result['object']['value']);
+  drush_log(dt('Processing @pid.', array(
+    '@pid' => $object->id,
+  )));
+  // Get the POLICY and compare it to the RELS-INT statements.
+  $xacml = new IslandoraXacml($object);
+  $to_update = FALSE;
+  $datastream_data = $xacml->datastreamRule->getRuleArray();
+  foreach ($datastream_data['dsids'] as $dsid) {
+    // Only check DSIDs that exist on the object given the nature of how the API
+    // writes RELS-INT statements when that datastream gets ingested.
+    if (isset($object[$dsid])) {
+      // Ensure the users and roles match up.
+      foreach ($datastream_data['users'] as $user) {
+        $user_rels = $object[$dsid]->relationships->get(ISLANDORA_RELS_INT_URI, 'isViewableByUser', $user, RELS_TYPE_PLAIN_LITERAL);
+        if (!$user_rels) {
+          $to_update = TRUE;
+          break 2;
+        }
+      }
+      foreach ($datastream_data['roles'] as $role) {
+        $role_rels = $object[$dsid]->relationships->get(ISLANDORA_RELS_INT_URI, 'isViewableByRole', $role, RELS_TYPE_PLAIN_LITERAL);
+        if (!$role_rels) {
+          $to_update = TRUE;
+          break 2;
+        }
+      }
+    }
+  }
+  if ($to_update) {
+    drush_log(dt('The POLICY for @pid is missing RELS-INT statements.', array(
+      '@pid' => $object->id,
+    )));
+    if ($commit) {
+      drush_log(dt('Updating the RELS-INT statements for @pid.', array(
+        '@pid' => $object->id,
+      )));
+      $xacml->writeBackToFedora();
+    }
+  }
+  $sandbox['progress'] = min($sandbox['total'], $sandbox['progress'] + 1);
+  $context['finished'] = $sandbox['progress'] / $sandbox['total'];
+  $context['message'] = t('Updated @progress of @total.', array(
```

Comment:
Given the separate `$commit` bit... doesn't quite make sense to say "Updated" if nothing was actually done to the object...